utils/patientmatching/rule_based_patient_matcher.bal

```diff
@@ -0,0 +1,274 @@
+        };
+        foreach int i in 0 ... patientArray.length()-1 {
+            r4:BundleEntry bundleEntry = {
+                'resource: patientArray[i],
+                search: bundleEntrySearch
+            };
+            patients.push(bundleEntry);
+        }
+        return patients;
+    }
+
+    isolated function getMPIData(r4:Patient patient, RulesRecord rulesTable, ConfigurationRecord config) returns stream<record {}, sql:Error?>|error {
+        sql:Client|error dbClient = self.getMpiDbClient(config);
+        if dbClient is error {
+            return throwFHIRError("Error occurred while getting the database client to access MPI.", cause = dbClient);
+        }
+        error|string qry = self.getSQLQuery(patient, rulesTable, config);
+        if qry is error {
+            return throwFHIRError("Error occurred while generating SQL query.", cause = qry);
+        }
+        sql:ParameterizedQuery queryString = ``;
+        queryString.strings = [qry];
+        stream<record {}, sql:Error?> resultStream = dbClient->query(queryString);
+        return resultStream;
+    }
+
+    isolated function getMPIConfigData(ConfigurationRecord config) returns MPIDbConfig|error {
+        json|error masterPatientIndexHost = config?.masterPatientIndexHost;
+        json|error masterPatientIndexPort = config?.masterPatientIndexPort;
+        json|error masterPatientIndexDb = config?.masterPatientIndexDb;
+        json|error masterPatientIndexDbUser = config?.masterPatientIndexDbUser;
+        json|error masterPatientIndexDbPassword = config?.masterPatientIndexDbPassword;
+        if masterPatientIndexHost is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexHost from the config.json file.", cause = masterPatientIndexHost);
+        }
+        if masterPatientIndexPort is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexPort from the config.json file.", cause = masterPatientIndexPort);
+        }
+        if masterPatientIndexDb is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexDb from the config.json file.",cause = masterPatientIndexDb);
+        }
+        if masterPatientIndexDbUser is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexDbUser from the config.json file.",cause = masterPatientIndexDbUser);
+        }
+        if masterPatientIndexDbPassword is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexDbPassword from the config.json file.", cause = masterPatientIndexDbPassword);
+        }
+        MPIDbConfig mpiDbConfig = {
+            host: <string>masterPatientIndexHost,
+            port: <int>masterPatientIndexPort,
+            database: <string>masterPatientIndexDb,
+            username: <string>masterPatientIndexDbUser,
+            password: <string>masterPatientIndexDbPassword
+        };
```

Comment:
remove casting, rename the variables to match with field name. eg: rename `masterPatientIndexHost` to `host` then u can just do `return {host, port, database, username, password}`

Reply:
Check other places as well

Reply:
Fixed in 6b572e1a82f29a631673acb2710d7f455ad43f82